src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorSourceDocument.cs

```diff
@@ -295,6 +249,37 @@ public static RazorSourceDocument Create(string content, Encoding encoding, Razo
             throw new ArgumentNullException(nameof(properties));
         }
 
-        return new StringSourceDocument(content, encoding, properties);
+        var sourceText = SourceText.From(content, encoding, checksumAlgorithm: SourceHashAlgorithm.Sha256);
+        return new RazorSourceDocument(sourceText, properties);
+    }
+
+    /// <summary>
+    /// Creates a <see cref="RazorSourceDocument"/> from the specified <paramref name="text"/>.
+    /// </summary>
+    /// <param name="content">The source text.</param>
+    /// <param name="properties">Properties to configure the <see cref="RazorSourceDocument"/>.</param>
+    /// <returns>The <see cref="RazorSourceDocument"/>.</returns>
+    public static RazorSourceDocument Create(SourceText text, RazorSourceDocumentProperties properties)
+    {
+        if (text == null)
+        {
+            throw new ArgumentNullException(nameof(text));
+        }
+
+        if (properties == null)
+        {
+            throw new ArgumentNullException(nameof(properties));
+        }
+
+        return new RazorSourceDocument(text, properties);
+    }
+
+    private RazorSourceDocument(SourceText sourceText, RazorSourceDocumentProperties properties)
+    {
+        Debug.Assert(sourceText != null);
+        Debug.Assert(properties != null);
```

Comment:
These seem redundant in private annotated code?

Reply:
Yeah, you're probably right.